### DISCUSSION TITLE
Improve database charset/collation validate warnings

### DISCUSSION THREAD

**Discussion by ArthurYidi** [directus/directus#18470] [Feature Requests]
### Summary

Tables using JSON Data Type (including Directus and user created tables) cause the following warning: 
```
[19:13:17.567] WARN: Some tables and columns do not match your database's default collation (utf8mb4_general_ci):
		- Table "directus_revisions": "utf8mb4_general_ci"
		  - Column "data": "utf8mb4_bin"
		  - Column "delta": "utf8mb4_bin"
...
```

Directus runs `validateDatabaseCharset` every time it initializes database connections (`getDatabase`), confusing developers with long warning messages. The reason these columns don't match `utf8mb4_general_ci` is because MariaDB defines `attr JSON` as `attr LONGTEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_bin ...`. We believe these warnings are safe to ignore.

We are proposing two solutions (see alternatives).

### Motivation

Remove developer confusion when viewing logs using MariaDB.

### Detailed Design

See Alternatives.

### Must Have:
- Handle JSON data types in validation, so warnings are more accurate.

### Drawbacks

- More edge cases to handle

### Alternatives

- Silence `longtext` columns with `utf8mb4_bin` for MariaDB (as implemented in https://github.com/directus/directus/pull/14787)

OR

- Provide an option to disable `validateDatabaseCharset` completely

### Adoption Strategy

No breaking changes.

### Background

- https://github.com/directus/directus/discussions/11786
- https://mariadb.com/resources/blog/using-json-in-mariadb/

**Comment by iksent** [comment]
This issue is really annoying, I am experiencing it at my every project
<img width="873" alt="image" src="https://github.com/directus/directus/assets/32956521/2158d345-e06f-4fd1-9232-65d5797b6a08">

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by martin-braun** [reply]
I'd rather consider this a bug than a feature request tbh.
